fastly.py

```diff
@@ -11,119 +11,204 @@
+
+    if args.start_time:
+        start_time = args.start_time
+    else:
+        # Select reasonable default intervals for the query. For minutely
+        # interval, we return hours worth of data; for hourly, we return
+        # one day; and for a daily interval, we'll return a month (30 days).
+        if interval == 'minute':
+            start_time = now - 3600  # 60 * 60
+        elif interval == 'hour':
+            start_time = now - 86400  # 24 * 60 60
+        elif interval == 'day':
+            start_time = now - 3456000  # 30 * 24 * 60 * 60
+        else:
+            start_time = now - 3600
+    # endregion Setting the from and to timestamps
+
+    service = None
     if args.service:
-        try:
-            all_services = {args.service: all_services[args.service]}
-        except:
-            print("unknown service: %s" % args.service)
+        service = get_service_by_name(args.service)
+        if not service:
+            print('Unknown Service: {0:s}'.format(args.service))
             sys.exit(1)
+        all_services = {service: args.service}
+    else:
+        all_services = get_services()
 
-    # Always set the end time to now - 30 mintues to not get rate limted,
-    # if you more recent data, you need to specify start and end time
-    # using the -f and -t parameter
-    now=int(time.time())
-    endtime = now - 1900;
+    string = GRAPHITE_PREFIX + '.{service}.{region}.{{value}}'
+    regions = get_regions()
+    for region in regions:
+        try:
+            stats_data = get_service_data(service=service, region=region,
+                                          cfrom=start_time,
+                                          to=end_time, interval=interval)
+        except BaseException as e:
+            print(e)
+            continue
+
+        for service, data in stats_data.items():
+            if service not in all_services:
+                continue
 
-    # select reasonable default intervals for the query,
-    # for a minutely interval we return an hours worth of data
-    # for an hourly we return one day and
-    # for a daily interval we'll retrun a months
-    interval = args.interval
-    if    interval == 'minute': starttime = now - 2500
-    elif  interval == 'hour':   starttime = now - 88201
-    elif  interval == 'day':    starttime = now - 2680201
+            service_name = all_services[service]
+            service_name = service_name.replace(' ', '_')
+            try:
+                output = string.format(service=service_name, region=region)
 
-    # use the provided start and end if present
-    if args.end_time: endtime = args.end_time
-    if args.start_time: starttime = args.start_time
+                for entry in data:
+                    for key in entry:
+                        value = format_key(entry, key)
+                        if not value:
+                            continue
+                        print(output.format(value=value))
```

Comment:
I prefer the "interesting" part in the "main" indention level. If, for whatever reasons, someone will add more functionality besides the `print`, then it is easier to read in my opinion. :wink: